nemo/collections/nlp/models/text_classification_model.py

```diff
@@ -132,40 +172,44 @@ def validation_epoch_end(self, outputs):
         """
         avg_loss = torch.stack([x['val_loss'] for x in outputs]).mean()
         # val_acc = sum([x['n_correct_pred'] for x in outputs]) / sum(x['n_pred'] for x in outputs)
-        # tensorboard_logs = {'val_loss': avg_loss, 'val_acc': val_acc}
-        return {'val_loss': avg_loss}  # , 'log': tensorboard_logs}
+        tensorboard_logs = {'val_loss': avg_loss}  # , 'val_acc': val_acc}
+        # return {'val_loss': avg_loss}  # , 'log': tensorboard_logs}
+        return {'val_loss': avg_loss, 'log': tensorboard_logs}
 
-    def setup_training_data(self, file_path, dataloader_params={}):
-        self._train_dl = self.__setup_dataloader(input_file=file_path, dataloader_params=dataloader_params)
+    def setup_training_data(self, train_data_config: Optional[DictConfig]):
+        # if 'shuffle' not in train_data_config:
+        #     train_data_config['shuffle'] = True
+        self._train_dl = self.__setup_dataloader_from_config(cfg=train_data_config)
 
-    def setup_validation_data(self, file_path, dataloader_params={}):
-        self._validation_dl = self.__setup_dataloader(input_file=file_path, dataloader_params=dataloader_params)
+    def setup_validation_data(self, val_data_config: Optional[DictConfig]):
+        self._validation_dl = self.__setup_dataloader_from_config(cfg=val_data_config)
 
-    def setup_test_data(self, file_path, dataloader_params={}):
-        self._test_dl = self.__setup_dataloader(input_file=file_path, dataloader_params=dataloader_params)
+    def setup_test_data(self, test_data_config: Optional[DictConfig]):
+        self._test_dl = self.__setup_dataloader(cfg=test_data_config)
 
-    def __setup_dataloader(
-        self, input_file, dataloader_params={},
-    ):
+    def __setup_dataloader_from_config(self, cfg: TextClassificationDataConfig):
+        input_file = f'{self.data_dir}/{cfg.prefix}.tsv'
+        # TODO: check that file exists
         dataset = TextClassificationDataset(
             input_file=input_file,
             tokenizer=self.tokenizer,
-            max_seq_length=dataloader_params.get("max_seq_length", 512),
-            num_samples=dataloader_params.get("num_samples", -1),
-            shuffle=dataloader_params.get("shuffle", False),
-            use_cache=dataloader_params.get("use_cache", False),
+            max_seq_length=self.max_seq_length,
```

Comment:
Is it possible to not have such inputs in the config files? So the config file should always contain all the params?

Reply:
The config can have default values, either null or some other "default" value. However, all common arguments that are routinely modified during hyper parameters search should be kept there (even as default values), so that we don't need the `+` syntax constantly to add parameters via hydra, and we have auto complete in the command line.

Reply:
I think mandatory configs should go into config object ```TextClassificationModelConfig```

We can decide what makes sense to go into the config.yaml. Just remember CLI overrides config.yaml overrides config object.